src/parameterizations/lateral/MOM_thickness_diffuse.F90

```diff
@@ -993,10 +993,14 @@ subroutine thickness_diffuse_full(h, e, Kh_u, Kh_v, tv, uhD, vhD, cg1, dt, G, GV
       ! Solve an elliptic equation for the streamfunction following Ferrari et al., 2010.
       do I=is-1,ie
         if (G%mask2dCu(I,j)>0.) then
-          Sfn_unlim_u(I,:) = ( 1. + CS%FGNV_scale ) * Sfn_unlim_u(I,:)
+          do K=2,nz
+            Sfn_unlim_u(I,K) = (1. + CS%FGNV_scale) * Sfn_unlim_u(I,K)
```

Comment:
This is awesome 👏 thank you @marshallward 

Any reason why it can't be 
_vectorized_? e.g., 
x(index1_range, index2_range) =0. 

- Do you anticipate different behavior for different optimizations? 
- violates coding style?
- something else?

Reply:
Yes, I think `Sfn_unlim_[uv](2:nz) = ...` would have been OK here, other than that it's discouraged by the style guide (although this one might be permitted).

This next line is a very explicit break from the style guide:
```
call streamfn_solver(nz, c2_h_v(i,:), hN2_v(i,:), Sfn_unlim_v(i,:))
```
due to the sliced array inputs - and the implicit creation - but they are only 1D arrays so I figured let's leave it for another day.

I wouldn't expect vectorization to be affected in this case, but compilers always have a surprise or two.